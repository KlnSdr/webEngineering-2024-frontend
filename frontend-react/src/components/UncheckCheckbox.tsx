import React from "react";
import Stack from "react-bootstrap/Stack";
import {Simulate} from "react-dom/test-utils";
import change = Simulate.change;
import SurveyService from "../services/SurveyService";
import {Survey} from "../types/Surveys";

interface UncheckCheckboxProps {
    voteChange: ()=> void;

}

const  UncheckCheckbox: React.FC<UncheckCheckboxProps> = ({ voteChange}) => {

    function getAllCheckboxes(): HTMLInputElement[] {
        return Array.from(document.querySelectorAll('input[type="checkbox"]')) as HTMLInputElement[];
    }

    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
        const index = getAllCheckboxes()
        for (let i = 0; i < index.length; i++) {
            if (index[i].checked) {

                index[i].checked = false;
            }
        }
        event.target.checked = true;
        voteChange();
    }

    return (
         <div className="form-check m-lg-4">
            <input className="form-check-input" type="checkbox" value="" id="flexCheckDefault"  onChange={handleChange}/>
            <label className="form-check-label" htmlFor="flexCheckChecked"></label>
         </div>

    );
}

export default UncheckCheckbox;