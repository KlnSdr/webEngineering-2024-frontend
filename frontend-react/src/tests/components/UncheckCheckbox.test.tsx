import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom/extend-expect';
import UncheckCheckbox from "../../components/UncheckCheckbox";

describe("UncheckCheckbox Component", () => {

    test("renders with initial value", () => {
        const voteChange = jest.fn();
        const {asFragment} = render(<UncheckCheckbox voteChange={voteChange} />);
        const checkbox = screen.getByRole("checkbox");
        expect(checkbox).not.toBeChecked();
        expect(asFragment()).toMatchSnapshot();
    });

    test("calls voteChange callback when checkbox is clicked", () => {
        const voteChange = jest.fn();
        render(<UncheckCheckbox voteChange={voteChange} />);
        const checkbox = screen.getByRole("checkbox");
        userEvent.click(checkbox);
        expect(checkbox).toBeChecked();
        expect(voteChange).toHaveBeenCalledTimes(1);
    });

    test(" check checkbox when clicked", () => {
        const voteChange = jest.fn();
        render(<UncheckCheckbox voteChange={voteChange} />);
        const checkbox = screen.getByRole("checkbox");
        userEvent.click(checkbox);
        expect(checkbox).toBeChecked();
    });
});