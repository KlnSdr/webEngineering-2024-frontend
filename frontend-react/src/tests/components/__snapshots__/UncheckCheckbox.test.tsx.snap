// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UncheckCheckbox Component renders with initial value 1`] = `
<DocumentFragment>
  <div
    class="form-check m-lg-4"
  >
    <input
      class="form-check-input"
      id="flexCheckDefault"
      type="checkbox"
      value=""
    />
    <label
      class="form-check-label"
      for="flexCheckChecked"
    />
  </div>
</DocumentFragment>
`;
